node/bindnode/api.go

```diff
@@ -47,11 +49,78 @@ func Prototype(ptrType interface{}, schemaType schema.Type) schema.TypedPrototyp
 		if schemaType == nil {
 			schemaType = inferSchema(goType, 0)
 		} else {
-			verifyCompatibility(make(map[seenEntry]bool), goType, schemaType)
+			verifyCompatibility(cfg, make(map[seenEntry]bool), goType, schemaType)
 		}
 	}
 
-	return &_prototype{schemaType: schemaType, goType: goType}
+	return &_prototype{cfg: cfg, schemaType: schemaType, goType: goType}
+}
+
+type converter struct {
+	kindType reflect.Type
+	fromFunc reflect.Value // func(Kind) (*Custom, error)
+	toFunc   reflect.Value // func(*Custom) (Kind, error)
+}
+
+type config struct {
```

Comment:
do you expect additional config members, or could you simplify this to:
```
type config map[reflect.Type]converter
```

Reply:
:thumbsup: